src/odemis/driver/zeiss.py

```diff
@@ -513,66 +565,99 @@ def _refreshPosition(self):
 
     def _doMoveRel(self, future, shift):
         """
-        move by the shift
-        shift (float): unit m
+        prepare the stage for a relative move using the values in shift
+        :param future: a future that can be used to manage a move
+        :param shift (dict): contains float position values in m or deg
         """
-        x, y, z, _ = self.parent.GetStagePosition()
+        if len(set(shift.keys()).intersection({"rx", "rz", "m"})) > 0:
```

Comment:
This works fine, as any non-empty set is considered `True`:
```suggestion
        if set(shift.keys()).intersection({"rx", "rz", "m"}):
```